### PR TITLE
imp: do not store total escrow when amount is zero

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: #3549

### Commit Message / Changelog Entry

```bash
N/A
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/transfer/keeper/keeper.go [hunk 1]
```diff
@@ -169,8 +169,14 @@ func (k Keeper) SetTotalEscrowForDenom(ctx sdk.Context, coin sdk.Coin) {
...
 	store := ctx.KVStore(k.storeKey)
-	bz := k.cdc.MustMarshal(&sdk.IntProto{Int: coin.Amount})
-	store.Set(types.TotalEscrowForDenomKey(coin.Denom), bz)
+	key := types.TotalEscrowForDenomKey(coin.Denom)
+
+	if coin.Amount.IsZero() {
+		store.Delete(key)
+	} else {
+		bz := k.cdc.MustMarshal(&sdk.IntProto{Int: coin.Amount})
+		store.Set(types.TotalEscrowForDenomKey(coin.Denom), bz)
```
Comment (DimitrisJim) [inline modules/apps/transfer/keeper/keeper.go:178]:
```suggestion
		store.Set(key, bz)
```

File: modules/apps/transfer/keeper/keeper.go [hunk 2]
```diff
@@ -169,8 +169,14 @@ func (k Keeper) SetTotalEscrowForDenom(ctx sdk.Context, coin sdk.Coin) {
...
-	bz := k.cdc.MustMarshal(&sdk.IntProto{Int: coin.Amount})
-	store.Set(types.TotalEscrowForDenomKey(coin.Denom), bz)
+	key := types.TotalEscrowForDenomKey(coin.Denom)
+
+	if coin.Amount.IsZero() {
+		store.Delete(key)
+	} else {
+		bz := k.cdc.MustMarshal(&sdk.IntProto{Int: coin.Amount})
+		store.Set(key, bz)
+	}
```
Comment (damiannolan) [inline modules/apps/transfer/keeper/keeper.go:179]: nit: we could return early and omit the `else`, maybe its more "idiomatic" go.

I don't feel too strongly about it here tho

```suggestion
	if coin.Amount.IsZero() {
		store.Delete(key)
		return
	}
	
	bz := k.cdc.MustMarshal(&sdk.IntProto{Int: coin.Amount})
	store.Set(key, bz)
```

**Review by damiannolan — APPROVED** [review]
Awesome, thanks @crodriguezvega 🚀

File: modules/apps/transfer/keeper/keeper.go [hunk 3]
```diff
@@ -169,8 +169,15 @@ func (k Keeper) SetTotalEscrowForDenom(ctx sdk.Context, coin sdk.Coin) {
 	}
 
 	store := ctx.KVStore(k.storeKey)
+	key := types.TotalEscrowForDenomKey(coin.Denom)
+
+	if coin.Amount.IsZero() {
+		store.Delete(key)
+		return
+	}
```
Comment (colin-axner) [inline modules/apps/transfer/keeper/keeper.go:177]: needs code documentation on why we do this. I like having godoc and in-code comment

**Review by colin-axner — COMMENTED** [review]
LGTM, will ack with code documentation

**Pull request merged**